src/GraphQL.ApiTests/net6/GraphQL.approved.txt

```diff
@@ -1685,30 +1685,50 @@ namespace GraphQL.Types
     }
     public class EnumValueDefinition : GraphQL.Utilities.MetadataProvider, GraphQL.Types.IProvideDeprecationReason, GraphQL.Types.IProvideDescription
     {
-        public EnumValueDefinition() { }
+        public EnumValueDefinition(string name, object? value) { }
         public string? DeprecationReason { get; set; }
         public string? Description { get; set; }
-        public string Name { get; set; }
-        public object? Value { get; set; }
+        public string Name { get; }
+        public object? Value { get; }
```

Comment:
I would like to have `EnumerationGraphType<T>` scan for `GraphQLAttribute` instances and apply them to `EnumValueDefinition`.  The attribues should be able to set the name and/or value, description, deprecation reason, and metadata on the definition.  As such, the `Name` property should not be read-only, since the `EnumValueDefinition` will be passed to the attribute class to be modified.  The `value` could be read-only, I guess.

We should be able to do this:

```cs
[Name("SortOrderEnum")]
public enum SortOrders
{
    [Description("Last edit date")]
    [Name("LAST_EDIT_DATE")]
    LastModifiedDate = 1,
}
```

Reply:
OK, `Name` is read/write now, see comment:
>  //TODO: important to set this property only BEFORE EnumValuesBase.AddValue called